lib/src/main/java/uk/gov/di/data/lep/library/dto/deathnotification/DeathNotificationSetMapper.java

```diff
@@ -41,6 +44,66 @@ public static DeathNotificationSet generateDeathNotificationSet(GroJsonRecord gr
         );
     }
 
+    @Tracing
+    public static DeathNotificationSet generateMinimisedDeathNotificationSet(
+        DeathNotificationSet oldDeathNotificationSet,
+        List<EnrichmentField> enrichmentFields
+    ) {
+        if (oldDeathNotificationSet.events().deathRegistrationEvent() != null) {
+            var minimisedDeathRegistrationEvent = minimiseNewDeathRegistration(
+                oldDeathNotificationSet.events().deathRegistrationEvent(),
+                enrichmentFields
+            );
+            return new DeathNotificationSet(
+                oldDeathNotificationSet.aud(),
+                new DeathRegistrationEventMapping(minimisedDeathRegistrationEvent, null),
```

Comment:
I stand by my statement before of differentiating these by types more - Jackson has some built in type discrimination stuff, e.g. https://stackoverflow.com/a/44124343 but I'm not _too_ fussed right now.